src/transactions/test/SetTrustLineFlagsTests.cpp

```diff
@@ -287,3 +288,767 @@ TEST_CASE("set trustline flags", "[tx][settrustlineflags]")
+                    getRevokeBalanceID(acc1, native, poolNative1, 0);
+                auto acc1Cur2BalanceID =
+                    getRevokeBalanceID(acc1, cur2, poolNative1, 1);
+                auto acc1NativeBalanceID =
+                    getRevokeBalanceID(acc1, native, poolNative1, 1);
+
+                REQUIRE_THROWS_AS(
+                    acc1.claimClaimableBalance(rootNativeBalanceID),
+                    ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+                REQUIRE_THROWS_AS(
+                    acc1.claimClaimableBalance(acc1NativeWrongIndexBalanceID),
+                    ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+                REQUIRE_THROWS_AS(acc1.claimClaimableBalance(acc1Cur2BalanceID),
+                                  ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+
+                acc1.claimClaimableBalance(acc1NativeBalanceID);
+            }
+
+            SECTION("claimable balance created for issuer")
+            {
+                auto acc1Usd = makeAsset(acc1, "USD1");
+                auto share1Usd = makeChangeTrustAssetPoolShare(
+                    cur1, acc1Usd, LIQUIDITY_POOL_FEE_V18);
+                auto pool1Usd = xdrSha256(share1Usd.liquidityPool());
+
+                acc1.changeTrust(cur1, 10);
+                root.pay(acc1, cur1, 10);
+
+                acc1.changeTrust(share1Usd, 10);
+
+                acc1.liquidityPoolDeposit(pool1Usd, 10, 10, Price{1, 1},
+                                          Price{1, 1});
+
+                checkLiquidityPool(*app, pool1Usd, 10, 10, 10, 1);
+
+                root.denyTrust(cur1, acc1, flagOp);
+
+                auto cur1BalanceID =
+                    getRevokeBalanceID(root, cur1, pool1Usd, 0);
+                auto usdBalanceID =
+                    getRevokeBalanceID(root, acc1Usd, pool1Usd, 0);
+
+                root.allowTrust(cur1, acc1);
+
+                REQUIRE(!acc1.hasTrustLine(
+                    changeTrustAssetToTrustLineAsset(share1Usd)));
+
+                // Pool should be deleted since the last pool share
+                // trustline was deleted
+                {
+                    LedgerTxn ltx(app->getLedgerTxnRoot());
+                    REQUIRE(!loadLiquidityPool(ltx, pool1Usd));
+                }
+
+                auto cur1Tl = acc1.loadTrustLine(cur1);
+                REQUIRE(getTrustLineEntryExtensionV2(cur1Tl)
+                            .liquidityPoolUseCount == 0);
+
+                REQUIRE(acc1.getTrustlineBalance(cur1) == 0);
+                acc1.claimClaimableBalance(cur1BalanceID);
+                REQUIRE(acc1.getTrustlineBalance(cur1) == 10);
+
+                // A claimable balance was not created for usd because acc1 is
+                // the issuer
+                REQUIRE_THROWS_AS(acc1.claimClaimableBalance(usdBalanceID),
+                                  ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+            }
+
+            SECTION("revoke from 0 balance pool share trustline")
```

Comment:
Probably worth checking the case where only one balance is 0 due to rounding.

Reply:
That test is already on our todo list of CAP-0038 tests. I think we can implement it in a subsequent PR to avoid holding this PR up. Let me know if you disagree.

Reply:
Yes that's fine.